sdks/go/pkg/beam/core/runtime/graphx/translate.go

```diff
@@ -420,33 +490,58 @@ func (m *marshaller) expandCrossLanguage(namedEdge NamedEdge) string {
 		// map consumers of these outputs to the expanded transform's outputs.
 		outputs := make(map[string]string)
 		for i, out := range edge.Output {
-			m.addNode(out.To)
+			if _, err := m.addNode(out.To); err != nil {
+				return "", errors.Wrapf(err, "Failed to expand cross language transform for edge: %v", namedEdge)
+			}
 			outputs[fmt.Sprintf("i%v", i)] = nodeID(out.To)
 		}
 		transform.Outputs = outputs
-		transform.EnvironmentId = ExpandedTransform(edge.External.Expanded).EnvironmentId
+		environment, err := ExpandedTransform(edge.External.Expanded)
+		if err != nil {
+			return "", errors.Wrapf(err, "Failed to expand cross language transform for edge: %v", namedEdge)
+		}
+		transform.EnvironmentId = environment.EnvironmentId
 	}
 
 	m.transforms[id] = transform
-	return id
+	return id, nil
 }
 
-func (m *marshaller) expandCoGBK(edge NamedEdge) string {
+func (m *marshaller) expandCoGBK(edge NamedEdge) (string, error) {
 	// TODO(BEAM-490): replace once CoGBK is a primitive. For now, we have to translate
 	// CoGBK with multiple PCollections as described in cogbk.go.
 
 	id := edgeID(edge.Edge)
-	kvCoderID := m.coders.Add(MakeKVUnionCoder(edge.Edge))
-	gbkCoderID := m.coders.Add(MakeGBKUnionCoder(edge.Edge))
+	kvCoder, err := MakeKVUnionCoder(edge.Edge)
+	if err != nil {
+		return "", errors.Wrapf(err, "Fail to expand CoGBK transform for edge: %v", edge)
```

Comment:
There's a lot of repetition in this function around the error handler. We can reduce that.

Make an anonymous helper function at the top of the function:
```
handleErr := func(err) (string, error) {
		return "", errors.Wrapf(err, "failed to expand CoGBK transform for edge: %v", edge)
}
```
And then replace the repeated return lines with:
```
  return handleErr(err)
```

Reply:
thanks, this is a good catch, I also create a anonymous function to return errors from _addMultiEdge_ method rather than create a named helper function for the same purpose.